src/Microsoft.AspNetCore.Http.Abstractions/HostString.cs

```diff
@@ -270,43 +327,43 @@ public override int GetHashCode()
         /// <summary>
         /// Parses the current value. IPv6 addresses will have brackets added if they are missing.
         /// </summary>
-        private void GetParts(out string host, out string port)
+        private static void GetParts(StringSegment value, out StringSegment host, out StringSegment port)
         {
             int index;
             port = null;
             host = null;
 
-            if (string.IsNullOrEmpty(_value))
+            if (StringSegment.IsNullOrEmpty(value))
             {
                 return;
             }
-            else if ((index = _value.IndexOf(']')) >= 0)
+            else if ((index = value.IndexOf(']')) >= 0)
             {
                 // IPv6 in brackets [::1], maybe with port
-                host = _value.Substring(0, index + 1);
+                host = value.Subsegment(0, index + 1);
 
-                if ((index = _value.IndexOf(':', index + 1)) >= 0)
+                if ((index = value.IndexOf(':', index + 1)) >= 0)
```

Comment:
I know this logic didn't change, but why call IndexOf here? Shouldn't the ':' immediately follow the ']' if it's present at all.